src/pages/workspace/WorkspaceInviteMessagePage.js

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Pressable, View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import Str from 'expensify-common/lib/str';
+import lodashGet from 'lodash/get';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import styles from '../../styles/styles';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as Policy from '../../libs/actions/Policy';
+import TextInput from '../../components/TextInput';
+import MultipleAvatars from '../../components/MultipleAvatars';
+import CONST from '../../CONST';
+import * as Link from '../../libs/actions/Link';
+import Text from '../../components/Text';
+import withPolicy, {policyPropTypes, policyDefaultProps} from './withPolicy';
+import * as ReportUtils from '../../libs/ReportUtils';
+import ROUTES from '../../ROUTES';
+import * as Localize from '../../libs/Localize';
+import Form from '../../components/Form';
+
+const personalDetailsPropTypes = PropTypes.shape({
+    /** The login of the person (either email or phone number) */
+    login: PropTypes.string.isRequired,
+
+    /** The URL of the person's avatar (there should already be a default avatar if
+    the person doesn't have their own avatar uploaded yet) */
+    avatar: PropTypes.string.isRequired,
+
+    /** This is either the user's full name, or their login if full name is an empty string */
+    displayName: PropTypes.string.isRequired,
+});
+
+const propTypes = {
+
+    /** All of the personal details for everyone */
+    personalDetails: PropTypes.objectOf(personalDetailsPropTypes),
+
+    invitedMembersDraft: PropTypes.arrayOf(PropTypes.string),
+
+    /** URL Route params */
+    route: PropTypes.shape({
+        /** Params from the URL path */
+        params: PropTypes.shape({
+            /** policyID passed via route: /workspace/:policyID/invite-message */
+            policyID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    ...policyPropTypes,
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    ...policyDefaultProps,
+    personalDetails: {},
+    invitedMembersDraft: [],
+};
+
+class WorkspaceInviteMessagePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.onSubmit = this.onSubmit.bind(this);
+        this.validate = this.validate.bind(this);
+        this.openPrivacyURL = this.openPrivacyURL.bind(this);
+        this.state = {
+            welcomeNote: this.getWelcomeNote(),
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            !(prevProps.preferredLocale !== this.props.preferredLocale
```

Comment:
I believe this means we want to pass this condition when the prev and current value of `preferredLocale` is the same? Can we just do `prevProps.preferredLocale === this.props.preferredLocale` to check?

Reply:
No. Here we want to fail if condition only when locale is changed and the welcome text is unchanged. So that welcome note will be updated only when this condition fails